R/utility.R

```diff
@@ -49,64 +49,144 @@ linuxWrapCommands <- function(commands = c()) {
   commandLine
 }
 
-#' Get a list of job statuses from the given job ids
+#' Get a list of job statuses from the given filter
 #'
-#' @param jobIds A character vector of job ids
+#' @param filter A filter containing job state
 #'
 #' @examples
 #' \dontrun{
-#' getJobList(c("job-001", "job-002"))
+#' getJobList()
 #' }
 #' @export
-getJobList <- function(jobIds = c()) {
-  filter <- ""
+getJobList <- function(filter = NULL) {
+  filterClause <- ""
 
-  if (length(jobIds) > 1) {
-    for (i in 1:length(jobIds)) {
-      filter <- paste0(filter, sprintf("id eq '%s'", jobIds[i]), " or ")
+  if (!is.null(filter)) {
+    if (!is.null(filter$state)) {
+      filterClause <-
+        paste0(filterClause, sprintf("state eq '%s'", filter$state))
     }
-
-    filter <- substr(filter, 1, nchar(filter) - 3)
   }
 
   jobs <-
-    rAzureBatch::listJobs(query = list("$filter" = filter, "$select" = "id,state"))
-  print("Job List: ")
-
-  for (j in 1:length(jobs$value)) {
-    tasks <- rAzureBatch::listTask(jobs$value[[j]]$id)
-    count <- 0
-    if (length(tasks$value) > 0) {
-      taskStates <-
-        lapply(tasks$value, function(x)
-          x$state == "completed")
-
-      for (i in 1:length(taskStates)) {
-        if (taskStates[[i]] == TRUE) {
-          count <- count + 1
+    rAzureBatch::listJobs(query = list("$filter" = filterClause, "$select" = "id,state"))
+
+  id <- character(length(jobs$value))
+  state <- character(length(jobs$value))
+  status <- character(length(jobs$value))
+  failedTasks <- integer(length(jobs$value))
+  totalTasks <- integer(length(jobs$value))
+
+  if (length(jobs$value) > 0) {
+    for (j in 1:length(jobs$value)) {
+      id[j] <- jobs$value[[j]]$id
+      state[j] <- jobs$value[[j]]$state
+      taskCounts <-
+        rAzureBatch::getJobTaskCounts(jobId = jobs$value[[j]]$id)
+      failedTasks[j] <-
+        as.integer(taskCounts$failed)
+      totalTasks[j] <-
+        as.integer(taskCounts$active + taskCounts$running + taskCounts$completed)
+
+      completed <- as.integer(taskCounts$completed)
+
+      if (totalTasks[j] > 0) {
+        if (completed > 0) {
+          status[j] <-
+            sprintf("%s %%", ceiling(completed / totalTasks[j] * 100))
+        } else {
+          status[j] <- "No tasks were run"
```

Comment:
I don't think we need this. Showing 0% should be sufficient.

Reply:
removed.